spec/var-ref.test.sh

```diff
@@ -13,6 +13,12 @@ echo ref ${!a} ${a}
 ## BUG mksh stdout: ref a b
 ## N-I dash/zsh stdout-json: ""
 
+#### var ref: positional params
+set -- x y
+a=1; b=@; c=*
+printf "|%s" ${!a} ".${!b}." ".${!c}." ; echo "|"
```

Comment:
The `printf` is there for this fun feature:

````
$ help printf
...
    The format is re-used as necessary to consume all of the arguments.
```

So this pattern effectively prints the arguments with a delimiter between them, the `|` from the format string.

The `echo` isn't strictly necessary; it's just there to stick a delimiter on the end for symmetry.

I noticed `argv.py` later today! Definitely a useful kind of thing to have. The specific formatting feels pretty verbose to me, though -- the test file ends up with a lot of commas and quotes, and often backslashes, that are all part of the `argv.py` encoding rather than part of the values it's describing. So I think I'd want to try to find a notation that's closer to a no-op in more cases. I don't think this one is optimal but I think it is closer.